example-app/src/app/local-state-advanced/local-state-advanced.component.ts

```diff
@@ -0,0 +1,55 @@
+import { ChangeDetectionStrategy, Component, ChangeDetectorRef, OnDestroy, OnInit } from '@angular/core';
+import { ActionsSubject, Action } from '@ngrx/store';
+import { SetValueAction, Actions } from 'ngrx-forms';
+import { Subscription } from 'rxjs';
+
+import { reducer, LocalState, GetManufacturersAction, INITIAL_LOCAL_STATE, FORM_ID } from './local-state-advanced.reducer';
+
+
+@Component({
+  selector: 'ngf-local-state-advanced',
+  templateUrl: './local-state-advanced.component.html',
+  styleUrls: ['./local-state-advanced.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LocalStateAdvancedComponent implements OnInit, OnDestroy {
+
+  public localState: LocalState = INITIAL_LOCAL_STATE;
+
+  private subscription = new Subscription();
+
+  constructor(private actionsSubject: ActionsSubject, private cd: ChangeDetectorRef) {
+  }
+
+  ngOnInit() {
+    this.subscription = this.actionsSubject.subscribe(action => {
+      const updated = this.updateState(action);
+      if (updated) {
+        // since OnPush is used, need to trigger detectChanges
+        // when action from outside updates localState
+        this.cd.detectChanges();
+      }
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
+  }
+
+  handleFormAction(action: Actions<any>) {
+    this.updateState(action);
+
+    // trigger loading of new manufacturers list in effect
+    if (action.type === SetValueAction.TYPE && action.controlId === `${FORM_ID}.countryCode`) {
```

Comment:
Instead of `${FORM_ID}.countryCode` which is not refactoring safe, I would just use `this.localState.formState.controls.CountryCode.id`, even if it is longer